### ISSUE TITLE
Custom axios instance causing options not to work

### ISSUE THREAD

**Issue by BoKKeR** [Flyrell/axios-auth-refresh#11]
First off, awesome project. But I can't figure out how to use the options.

This is the best I could come up with and it dosent work at all :/

`createAuthRefreshInterceptor(client, refreshAuthLogic, {statusCodes: [ 401, 500, 400 ]});`

**Comment by Flyrell** [comment]
Hi Norbert, this should be the way to use them. Can you please share your complete code for the interceptor, so I can take a look?

**Comment by BoKKeR** [comment]
It might be working on some level as I get no error in the console nor the response only this:

```
function wrap() {
      var args = new Array(arguments.length);

      for (var i = 0; i < args.length; i++) {
        args[i] = arguments[i];
      }

      return fn.apply(thisArg, args);
    }
```

Here is the full code.
```
import axios from "axios"
import config from "../config"
import createAuthRefreshInterceptor from 'axios-auth-refresh';

const client = axios.create({
  //baseURL: config.apiUrl,
  headers: {
    "Content-type": "application/json"
  }
});

client.defaults.headers.get["Content-Type"] = "application/json";

let myToken = "";

const refreshAuthLogic = err => client.post(config.auth0Url, {
  client_id: config.clientId,
  client_secret: config.clientSecret,
  audience: config.clientAudience,
  grant_type: "client_credentials"
}).then(res => {
  localStorage.setItem('token', res.data.token);
  err.response.config.headers['Authorization'] = 'Bearer ' + res.data.token;
  return Promise.resolve();
});

createAuthRefreshInterceptor(client, refreshAuthLogic, {statusCodes: [ 401, 500, 400 ]});
//400 no header at all
//500 header but bad //should be 401

console.log("token: " + myToken);
client.get(config.apiUrl + "groups/5ca21cd1cb9134222e397f14",
  {
    data: {},
    //headers: {"Authorization": "Bearer " + ""}
  })
  .then(response => {
    console.log(response);
    //this.handler(response.data.access_token)
  })
  .catch(function (error) {
    console.log("error: " + error);
  });


export default client;

```

**Comment by BoKKeR** [comment]
I am getting the token, and I am pretty sure it gets applied to the request. Its only that the response is wrong for some reason.

**Comment by Flyrell** [comment]
That's weird, because what you're seeing seems to be an actual [axios builder function](https://npmdoc.github.io/node-npmdoc-axios/build/apidoc.html#apidoc.element.axios.axios), that needs to be executed. How are you calling the client?

**Comment by BoKKeR** [comment]
I am only using the snippet that is in the example above. It has the instance, the call. I tried to move the get request and use it in a different screen but it returns the same builder function. It seems to just fail to print the response. With that code alone you should be able to reproduce it if its reproducible.

**Comment by Flyrell** [comment]
I can't actually reproduce the issue. When it comes to the `statusCodes` option, I'm pretty confident that it works properly. I might write the unit tests for this bundle when I have a little time and I might find the bug that causes your issue (I think it might be something with the axios instance). I'll surely let you know if something comes up. In case you're interested, I'd really appreciate the contribution in case you find the problem first.

**Comment by mvanroon** [comment]
I'm experiencing the same issue:

```
// api.js
import axios from 'axios'
import createAuthRefreshInterceptor from 'axios-auth-refresh'
import { getURL } from './env'

const instance = axios.create({
  baseURL: getURL()
})

export const setAccessToken = token => {
  instance.defaults.headers.common.Authorization = `Bearer ${token}`
}
export const unsetAccessToken = token => {
  delete instance.defaults.headers.common.Authorization
}

const refreshAuthLogic = failedRequest =>
  instance.put('/auth/refresh').then(tokenRefreshResponse => {
    const accessToken = tokenRefreshResponse.data.access_token
    setAccessToken(accessToken)
    failedRequest.response.config.headers.Authorization = `Bearer ${accessToken}`
    return Promise.resolve()
  })

export default createAuthRefreshInterceptor(instance, refreshAuthLogic)

```

```
// profile.js
import api from './api'

export const loadProfile = () => async dispatch => {
  const response = await api.get('/user')
  
  console.log(response)
}
```

This logs the response when the token was valid, but when the request gets retried, it logs the axios builder function

Could it have to do with the fact that we're using axios.create() instead of the default axios instance?

**Comment by ArondeParon** [comment]
I'm also experiencing this same issue, using a custom Axios instance. At this point I still have not got a clue as to what is causing this issue.

**Comment by Flyrell** [comment]
Sorry guys, I'm having a few days off and haven't had time to look at the mailbox 'till now. I'm not sure what might be causing the issue, but as you mentioned earlier, that's probably the `axios.create`. If anyone has time to look at this, feel free to create a PR. I'll be more than happy to accept it.

**Comment by Flyrell** [comment]
Fixed in #27. Please, give it a try before I publish it as latest release (it's in beta now). 

```
npm install axios-auth-refresh@beta --save
```
